### PR TITLE
Golang compiler and example package

### PR DESCRIPTION
**Pull request by erasche**

### CODE REVIEW

File: packages/package_golang_1_5_1/tool_dependencies.xml [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+            <!--
+                Golang compilation requires knowing what architecutre you're
+                building for. Unless this is available in an environment
+                variable... we aren't going to build a generic version.
+
+                GOOS=linux GOARCH=ppc64 ./bootstrap.bash
+            -->
+        <!--</actions>-->
+        <action type="set_environment">
+          <environment_variable action="set_to" name="GOROOT">$INSTALL_DIR</environment_variable>
```
Comment (bgruening) [inline packages/package_golang_1_5_1/tool_dependencies.xml]: GOLANG_ROOT_PATH

File: test_repositories/golang_package/tool_dependencies.xml [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+            <actions>
+                <action type="download_by_url">https://github.com/erasche/bin-fasta/archive/415756dcd3573f330a39b073c9a314b840c189e0/master.zip</action>
+
+              <action type="set_environment_for_install">
+                  <repository name="package_golang_1_5_1" owner="iuc">
+                    <package name="golang" version="1.5.1" />
+                  </repository>
+              </action>
+
+              <action type="shell_command">
```
Comment (bgruening) [inline test_repositories/golang_package/tool_dependencies.xml]: Can you add comments here. Would be nice to have for an example package.

File: packages/package_golang_1_5_1/tool_dependencies.xml (hunk 1 above)
Comment (erasche) [inline packages/package_golang_1_5_1/tool_dependencies.xml]: `GOROOT` is [special to go](http://stackoverflow.com/questions/7970390/what-should-be-the-values-of-gopath-and-goroot). Is `GOLANG_ROOT_PATH` really necessary now that we have https://github.com/galaxyproject/galaxy/pull/564?

File: test_repositories/golang_package/tool_dependencies.xml (hunk 2 above)
Comment (erasche) [inline test_repositories/golang_package/tool_dependencies.xml]: Ah, yes, can comment this more. Good suggestion :)

File: packages/package_golang_1_5_1/tool_dependencies.xml (hunk 1 above)
Comment (bgruening) [inline packages/package_golang_1_5_1/tool_dependencies.xml]: Depends on the backward compatibility. If you think it's ok to assume that all users of golang will use an up-to-date version, this is fine.

Comment (erasche) [inline packages/package_golang_1_5_1/tool_dependencies.xml]: I think that I'm the only person who is will be using it in the near future (6 months-1 year), I don't know of any other bio-tools written in golang yet.

**Comment by bgruening** [comment]
Thanks @erasche!

**Pull request merged**